lib/billing/request.ts

```diff
@@ -131,6 +131,14 @@ async function requestRecurringPayment({
                   amount: billingConfig.amount,
                   currencyCode: billingConfig.currencyCode,
                 },
+                discount: {
+                  durationLimitInIntervals:
+                    billingConfig.discount?.durationLimitInIntervals,
+                  value: {
+                    amount: billingConfig.discount?.value?.amount,
+                    percentage: billingConfig.discount?.value?.percentage,
```

Comment:
What is the return value from Admin when both are set?  In other way, does the GraphQL mutation send back an error?

Reply:
Yes, it does.
`"The discount value can not be both a monetary amount and a percentage."`

Reply:
We could add some extra types, so that folks would get a type error if they were to try to add both. Something like this

```
export interface BillingConfigSubscriptionPlan extends BillingConfigPlan {
  interval: RecurringBillingIntervals;
  trialDays?: number;
  replacementBehavior?: BillingReplacementBehavior;
  discount?: BillingConfigSubscriptionPlanDiscountConfig;
}

interface BillingConfigSubscriptionPlanDiscount {
  durationLimitInIntervals?: number;
}

interface BillingConfigSubscriptionPlanDiscountAmount
  extends BillingConfigSubscriptionPlanDiscount {
  amount: number;
  percentage?: never;
}

interface BillingConfigSubscriptionPlanDiscountPercentage
  extends BillingConfigSubscriptionPlanDiscount {
  amount?: never;
  percentage: number;
}

interface BillingConfigSubscriptionPlanDiscountConfig {
  value:
    | BillingConfigSubscriptionPlanDiscountAmount
    | BillingConfigSubscriptionPlanDiscountPercentage;
}

```